README.md

```diff
@@ -2,31 +2,630 @@
+
+The `Person` class defines fields `name` and `age`, along with methods that access and mutate
+the fields.  
+```java
+public class Person {
+
+    private String name;
+    private int age;
+
+    public String getName() { return name; }
+    public void setName(String name) { this.name = name; }
+    public int getAge() { return age; }
+    public void setAge(int age) { this.age = age; }
+    public void celebrateBirthday() { age++; }
+    public String getWeekendPlans() { return "sleep late and relax all day"; }
+    @Override
+    public String toString() { return "name=" + name + ", age=" + age; }
+}
+
+```
+
+The subclass `Teacher` overrides `getWeekendPlans()`:
+
+```java
+public class Teacher extends Person {
+    @Override
+    public String getWeekendPlans() { return "grade homework assignments"; }
+}
+```
+
+The subclass `Student` adds a `favoriteSubject` field and get/set methods, and overrides `getWeekendPlans` and `toString`:
+
+```java
+public class Student extends Person {
+    private String favoriteSubject;
+
+    public String getFavoriteSubject() { return favoriteSubject; }
+    public void setFavoriteSubject(String favoriteSubject) { this.favoriteSubject = favoriteSubject; }
+
+    @Override
+    public String getWeekendPlans() { return "wake up early and study all day"; }
+
+    @Override
+    public String toString() { return super.toString() + ", favoriteSubject=" + favoriteSubject ; }
+}
+```
+
+The `Main` driver class calls the no-args constructors `Person()`, `Teacher()`,
+and `Student()`, and then call the various setter methods to initialize the object state:
+
+```java
+public class Main {
+    public static void main(String[] args) {
+        //Person instance
+        System.out.println("Creating Person");
+        Person person = new Person();
+        person.setName("Kai");
+        person.setAge(32);
+        System.out.println("Person state:" + person);  
+
+        //Teacher instance, inherits fields and methods from Person
+        System.out.println("Creating Teacher");
+        Teacher teacher = new Teacher();
+        teacher.setName("Tal");
+        teacher.setAge(40);
+        System.out.println("Teacher state:" + teacher);   
+
+        //Student instance, inherits fields and methods from Person
+        System.out.println("Creating Student");
+        Student student = new Student();
+        student.setName("Hao");
+        student.setAge(24);
+        student.setFavoriteSubject("Java programming");
+        System.out.println("Student state:" + student);   
+    }
+}
+```
+
+The program prints as output:
+
+```text
+Creating Person
+Person state:name=Kai, age=32
+Creating Teacher
+Teacher state:name=Tal, age=40
+Creating Student
+Student state:name=Hao, age=24, favoriteSubject=Java programming
+```
+
+## Default No-Args Constructors
+
+Recall that Java generates a no-args constructor if a class does not explicitly
+define a constructor.  Let's update each class to add a
```

Comment:
"...add a explicit no-args constructor..."

Change "a" to "an"

Reply:
fixed